File: crates/deslop/tests/fixtures/ast-golden-rust/Sample.rs
struct Thing<'a, T> {
    name: &'a str,
    value: T,
}

impl<'a, T> Thing<'a, T> {
    async fn probe(&self, count: i64) -> f64 {
        let marker: char = 'x';
        let flag: bool = true;
        let width: f64 = 1.5;
        let label = format!("{}", self.name);
        match count {
            0 => 0.0,
            _ => count as f64,
        }
    }
}

File: crates/deslop/tests/fixtures/ast-golden-dart/Sample.dart
import 'dart:math';

/// A documented class.
class Thing {
  final int count;
  const Thing(this.count);

  String probe(int value) {
    // dropped comment
    final marker = 'v$value';
    final flag = true;
    final width = 1.5;
    final nothing = null;
    final pair = (value, count);
    return switch (pair) {
      (0, _) => 'zero',
      (final a, final b) when a > b => 'first',
      _ => marker,
    };
  }
}
